### ISSUE TITLE
Indexing should maintain a stable number of delayed rule executions

### ISSUE THREAD

**Issue by bh9** [irods/irods_capability_indexing#45]
Indexing should create new rule executions only once the previous batch of rule executions haev been completed/removed. 
Our zone now has 8.2m documents in elasticsearch, but 6.5m rule executions. That means 6.5m files on disk on the indexing provider. If, for example, the batch size defined the maximum number of rules it created at once, this plugin would have a much smaller hit on the available resources on that provider

**Comment by trel** [comment]
`packedRei` files are going away in 4.2.9 - does that reduce this concern completely?

**Comment by bh9** [comment]
There's still the issue that 6.5m rule executions makes it much harder to see what else is in the list. For us, it probably does remove the concern since we don't use any other delayed rules on the indexed zones

**Comment by trel** [comment]
So is this a visibility issue?   Rather than a load issue?

Can GenQuery filter/find what you’re looking for?

**Comment by trel** [comment]
Could be implemented as an additional loop around the collection indexer that considers the outstanding queue length before pushing more jobs onto the queue.

**Comment by korydraughn** [comment]
- Code to limit the number of delay jobs was merged years ago
- iRODS 4.2.9 is EOL
- Delay rule information is stored in the catalog instead of REI files
- Enhancements for limiting the size of the in-memory delay queue have been implemented
- There have been no new reports regarding this issue

Please open a new issue if this problem occurs in more recent versions of iRODS.

Closing.
